ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs

```diff
@@ -314,26 +314,38 @@ private void BuildAssemblyMonikerMapping(ReflectionItem item)
 
         private void BuildExtensionMethods()
         {
-            if (_extensionMethods == null || _extensionMethods.Count == 0)
-            {
-                return;
-            }
-            ExtensionMethodsByMemberDocId = _extensionMethods.ToDictionary(ex => ex.MemberDocId);
-
+            _extensionMethods = new List<Member>();
             foreach (var m in MembersByUid.Values)
             {
-                if (!string.IsNullOrEmpty(m.DocId) && ExtensionMethodsByMemberDocId.ContainsKey(m.DocId))
+                if (!string.IsNullOrEmpty(m.DocId))
                 {
-                    m.IsExtensionMethod = true;
-                    ExtensionMethodsByMemberDocId[m.DocId].Uid = m.Uid;
-                    ExtensionMethodsByMemberDocId[m.DocId].ParentType = m.Parent;
+                    var thisParam = m.Parameters.Where(p => { return p.RefType == "this"; });
```

Comment:
we usually don't reuse .Where() like this because it's just a query and it will re-evaluate every time you use it. So we'd better do ` = m.Parameters.FirstOrDefault(p => p.RefType == "this");` here.